### DISCUSSION TITLE
Can you filter a column only in one direction?

### DISCUSSION THREAD

**Discussion by symbiogenesis** [Mimetis/Dotmim.Sync#780] [Q&A] [answered]
If I would like to filter only when retrieving data, but not when submitting, is that easy to accomplish?

**Comment by Mimetis** [comment]
I don't understand your scenario, can you explain a little bit more in details ?

**Comment by symbiogenesis** [comment]
I have a blob data column whose data is generated by the clients, and they need to submit it to the server, but don't want to get all of the blob data from all clients synced back, because that data could be large and use up too much disk space on the clients.

Trying to just ignore the column in both directions makes it difficult, because a full sync would need to complete before submitting the data to the server in a separate custom REST request... otherwise, it could not necessarily be associated to the record via a foreign key if that record isn't on the server yet.

Using an interceptor on the server to strip the data from the OnTableChangesSelecting event seemed like a valid approach, but I haven't been able to make that work yet. I also tried to use the Converters, but wasn't able to make that work either. Not sure what the right feature to use is.

**Comment by Mimetis** [comment, marked as answer]
Removing the value on the `OnTableChangesSelecting` will not help, because it will delete the value on the client, I think.
I think using interceptor may help, but we can find another solution, maybe.

Something I would try is to use **2** scopes:

- **1st scope** (let's say scope name "`all`") that contains all your tables and the table containing the blob has all columns without the blob column.
This scope is `Bidirectional` (this is the default value). 
You will have all the rows, without the blob, and even if a conflict occurs or whatever, the blob on the client will never been touched.
Be sure the blob column allows null

- Then I will do a **2nd scope** with only the table with the blob, with all columns, and I will setup the table to be `UploadOnly`
This scope will upload the blob to the server but will never come back.

I think it's worth a try 
Let me know if it's working or if you found another solution

**Comment by symbiogenesis** [comment]
Ok, that seemed to work. Thank you :)

Now I just need to figure out why some other thing is happening, where a particular table is not syncing. Maybe related to a .NET point release upgrade that I did.

**Reply by Mimetis** [reply]
If you are testing, be sure to always start from scratch (ie a database without any DMS metadatas)

**Reply by Mimetis** [reply]
You can use `DropAllAsync` to remove everything, by the way

**Reply by symbiogenesis** [reply]
Thanks. I was using SQLite with EF Core 3.1.27 and upgraded to EF Core 3.1.28, and it definitely broke my syncing, even with newly generated database and DMS metadata. The top level table of a parent-child hierarchy didn't upload or download. Downgrading fixes it.

**Reply by Mimetis** [reply]
hum, that's strange (and weird) ...

**Comment by symbiogenesis** [comment]
Continuing along with this kind of thinking, I now have another use for this SyncDirection feature.

Basically, I can add a Logs table as UploadOnly;

In this case, I can have persistent local logging that can be saved even if there is no network connection, but it will still upload them later. And it won't try to download any of the logs from the server.

This is very important and cool. Thanks again! Great tool.

**Reply by Mimetis** [reply]
Indeed, that` s an accurate example of multi scopes sync 👍
